Some updates

Protonull opened the pull request.

Protonull committed:
PotionNames

Protonull committed:
Forgot about this

Protonull committed:
Merge pull request #3 from CivClassic/master

Keeping up

Protonull committed:
AikarCommandManager

Allow developers to create a command manager that doesn't automatically register commands. This is useful if you want to keep the same instance.

Protonull committed:
Minor cleanup

Protonull committed:
ACivMod - Javadoc Additions

Protonull committed:
PagedGUI - Fixed GUI registration

Protonull committed:
NBTCache addition

New class that simplifies direct encoding to and decoding from nbt, acting as an interface for more complex data. This differs in how NBTCompound is an interface as that provides the means of setting a string to a key for example, whereas the extended class is meant to provide getters and setters along the lines of setMaterial() for example. The setter will encode the correct data directly onto the NBT instead of storing it in a field to be serialised later.

Protonull committed:
Test :: ExampleSerializable swapped over to NBTCache

Protonull committed:
General move to slf4j logging

Protonull committed:
Update NBT Serialization tests

Protonull committed:
NBTCompound now validatable

Protonull committed:
Update to NBTCompound

Non-primitive values will now default to null. Class was also re-organised to group the array functions together. Clear now use a reflected clear, which is giving me some thoughts.

Protonull committed:
NBTCompound - Nevermind on the null default

In reality it just makes decoding frustrating, so I've added a check for NBT types.

Protonull committed:
Fixed test

Protonull committed:
Use Google Strings not jopsimple

Protonull committed:
Added EnumUtils

Protonull committed:
Small addition to supported colour tags

Protonull committed:
Case sensitivity

Protonull committed:
Remove unnecessary this usages

Protonull committed:
Deprecated manual NBTSerialization registration

Protonull committed:
Remove the expectation for NBTSerializable classes to throw a serialization exception

Protonull committed:
Javadoc

Protonull committed:
Javadoc - adds default values

Protonull committed:
Better than Object.equals() check

Protonull committed:
New Iteration API

Protonull committed:
small bugfix

Maxopoly commented on src/main/java/vg/civcraft/mc/civmodcore/util/Iteration.java:
```diff
@@ -169,4 +170,23 @@
+	 * @param entry The map entry itself.
+	 * @return Returns true if the entry is considered valid.
+	 */
+	public static boolean validEntry(Map.Entry<?, ?> entry) {
```
This seems really unneccessary given that Entries should only be returned by iterating entry sets, which will never give a null Entry and that none of the common Map implementations (HashMap, TreeMap and EnumMap) allow null keys.

Isn't this just a null check on the value?

Protonull replied:
Maybe I should've put into the javadoc that it's meant to be a stream filter rather than really anything else. It's akin to `.filter(Objects::nonNull)`

Protonull committed:
Added support for NBT lists of serializable classes, removing a lot of code that would otherwise be needed for serializing NBTSerializable classes.

Protonull committed:
Merge branch 'master' into civ14

Protonull committed:
Added potion material check

Added it to MaterialAPI in lieu of PotionAPI as I want to move away from fragmenting material checks away from MaterialAPI... it's the whole reason it exists as a gap-filler for Tags

Protonull committed:
POM :: Removed checkstyle source tag and prevent build failure

The source tag is invalid (read more: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#source)

Protonull committed:
NamespaceAPI

Because Bukkit is moving towards Namespaces, especially for enchantments.. would need to add something like this sooner or later.

Protonull committed:
Actually, ganna remove this

Added this in 09a9e2c (still in PR) and not even I use it.. not sure why I even made it tbh. Maybe it's a good example class? Rather not keep it around and have to maintain it.

Protonull committed:
Add support for NamespacedKey in EnchantAPI.getEnchantment()

Maxopoly commented on src/main/java/vg/civcraft/mc/civmodcore/api/NamespaceAPI.java:
```diff
@@ -0,0 +1,59 @@
+	 *     total length is longer than 256.
+	 */
+	@SuppressWarnings("deprecation")
+	public static NamespacedKey fromString(String key) {
```
Don't add a method and instantly deprecate it, just don't have it and tell people to git gud

Protonull replied:
I didn't D:

The `NamespacedKey`'s constructor that takes a namespace and a key is deprecated despite it being used internally for `NamespacedKey.minecraft(String)` and `NamespacedKey.randomKey()`. I'm suppressing the deprecation warning, not deprecating the API.

Maxopoly replied:
Oh wait i thought it was an `@Deprecated` annotation. Nevermind, carry on.

Protonull committed:
Why did I add these suppressions?

Protonull committed:
NamespaceAPI :: add more checks

Protonull committed:
Javadoc update

Protonull committed:
Make PagedGUI.clicked() package private

Protonull committed:
Javadoc + import cleanup

Protonull committed:
BlockAPI :: Fixed getOtherDoubleChestBlock()

+ some other useful functions.

Protonull committed:
As requested

Protonull committed:
Move these up
